src/tree/durableTaskScheduler/DurableTaskSchedulerEmulatorsWorkspaceResourceModel.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { type DurableTaskSchedulerWorkspaceResourceModel } from "./DurableTaskSchedulerWorkspaceResourceModel";
+import { localize } from "../../localize";
+import { treeUtils } from "../../utils/treeUtils";
+import {type DurableTaskSchedulerEmulator, type DurableTaskSchedulerEmulatorClient } from "./DurableTaskSchedulerEmulatorClient";
+import { DurableTaskSchedulerEmulatorWorkspaceResourceModel } from "./DurableTaskSchedulerEmulatorWorkspaceResourceModel";
+import { DurableTaskSchedulerErrorWorkspaceResourceModel } from "./DurableTaskSchedulerErrorWorkspaceResourceModel";
+import { parseError } from "@microsoft/vscode-azext-utils";
+
+export class DurableTaskSchedulerEmulatorsWorkspaceResourceModel implements DurableTaskSchedulerWorkspaceResourceModel {
+    private getEmulatorsTask: Promise<DurableTaskSchedulerEmulator[]>;
+
+    constructor(private readonly emulatorClient: DurableTaskSchedulerEmulatorClient) {
+    }
+
+    async getChildren(): Promise<DurableTaskSchedulerWorkspaceResourceModel[]> {
+        this.getEmulatorsTask = this.emulatorClient.getEmulators();
+
+        try {
+            const emulators = await this.getEmulatorsTask
+
+            return emulators.map(emulator => new DurableTaskSchedulerEmulatorWorkspaceResourceModel(emulator));
+        }
+        catch (error) {
+            return [new DurableTaskSchedulerErrorWorkspaceResourceModel(parseError(error).message)];
+        }
+    }
+
+    async getTreeItem(): Promise<TreeItem> {
+        this.getEmulatorsTask ??= this.emulatorClient.getEmulators();
+
+        const collapsibleState = await this.emulatorsExist()
+            ? TreeItemCollapsibleState.Expanded
+            : TreeItemCollapsibleState.Collapsed;
+
+        const treeItem = new TreeItem(localize('dtsEmulatorsLabel', 'Durable Task Scheduler Emulators'), collapsibleState);
+
+        treeItem.contextValue = 'azFunc.dts.emulators';
+        treeItem.iconPath = treeUtils.getIconPath('durableTaskScheduler/DurableTaskScheduler');
+
+        return treeItem;
+    }
+
+    id?: string | undefined;
+
+    private async emulatorsExist(): Promise<boolean> {
+        // Used any cached result that might exist...
+        this.getEmulatorsTask ??= this.emulatorClient.getEmulators();
```

Comment:
Is there any action that the user could take to change what `getEmulators` would output during a session? 

I ask because `getEmulatorsTask` never gets set to undefined, so this list would never get updated even if the user stopped an emulator. But I may be misunderstanding what "exists" means in this case.

Reply:
`getEmulatorsTask` is updated when the node is asked for children.  That said, the logic takes advantage of the fact that VS Code will refresh/recreate the parent node entirely in the event of user action (e.g. manually refreshing the window, or starting or stopping an emulator).  (See the provider's subscription to emulator changes.). The observation was that VS Code tended to call getTreeItem() quite often, for the same node, so the thought was to try to eliminate redundant calls to relatively slow container commands.